File: 0078-subsets/0078-subsets.java
class Solution {
  public List<List<Integer>> subsets(int[] nums) {
    List<List<Integer>> ans = new ArrayList();
    int ub = 1 << nums.length;
    for (int i = 0; i < ub; i++) {
      List<Integer> sub = new ArrayList<Integer>();
      for (int j = 0; j < nums.length; j++) {
        int p = i & 1 << j;
        if (p > 0) sub.add(nums[j]);
      }
      ans.add(sub);
    }
    return ans;
  }
}

File: 0078-subsets/0078-subsets.py
class Solution:
  def subsets(self, nums: List[int]) -> List[List[int]]:
    ans = []
    for i in range(2**len(nums)):
      sub = []
      for j in range(len(nums)):
        if i & 1 << j:
          sub.append(nums[j])
      ans.append(sub)
    return ans